param/parameterized.py

```diff
@@ -2611,7 +2759,38 @@ def serialize_value(self_, pname, mode='json'):
         serializer = Parameter._serializers[mode]
         return serializer.serialize_parameter_value(self_or_cls, pname)
 
-    def deserialize_parameters(self_, serialization, subset=None, mode='json'):
+    def deserialize_parameters(self_, serialization, subset=None, mode='json') -> dict:
```

Comment:
Now I better understand why I've never felt confident starting to use some of these methods.

They feel so "unsupported" when there is no useful docstring. And without types you have to experiment to figure out what goes in and comes out.